apisix/core/request.lua

```diff
@@ -94,7 +94,35 @@ function _M.get_remote_client_port(ctx)
         ctx = ngx.ctx.api_ctx
     end
     return tonumber(ctx.var.remote_port)
-  end
+end
+
+function _M.get_scheme(ctx)
+    if not ctx then
+        ctx = ngx.ctx.api_ctx
+    end
+    return ngx.var.scheme or ''
```

Comment:
that is not good, we should always use `ctx.var.****` which is better performance.